uswds/uswds
# USWDS - Core: Replace receptor library "keymap" with custom implementation

# Summary

**Reduced the JavaScript bundle size.** By optimizing the implementation of core component code, overall JavaScript sizes are reduced.

## Breaking change

This is not a breaking change.

## Related pull requests

Split from #5793, to simplify review.

Includes additional inline code comments based on review feedback from that pull request.

## Preview link

N/A

## Problem statement

See #5793

## Solution

Replace use of `receptor/keymap` with equivalent custom implementation.

## Testing and review

Verify unit tests pass:

```
npx gulp unitTests
```

For user acceptance testing, Modal is a good component to test for regressions, since its focus trap includes both simple key mapping (<kbd>Tab</kbd>) as well as modifier combinations (<kbd>Shift+Tab</kbd>).

1. Go to http://localhost:6006/?path=/story/components-modal--default
2. Click "Open default modal"
3. Press <kbd>Tab</kbd> or <kbd>Shift+Tab</kbd>
4. Observe that modal focus wraps as expected in focus trap
